packages/block-library/src/embed/edit.native.js

```diff
@@ -196,6 +196,14 @@ const EmbedEdit = ( props ) => {
 		isEditingURL,
 	] );
 
+	const onEditURL = useCallback( ( value ) => {
+		// The order of the following calls is important, we need to update the URL attribute before changing `isEditingURL`,
+		// otherwise the side-effect that potentially replaces the block when updating the local state won't use the new URL
+		// for creating the new block.
+		setAttributes( { url: value } );
+		setIsEditingURL( false );
+	}, [] );
```

Comment:
I've extracted the edit URL handler as now it's used in two places, the embed controls and the bottom sheet.